onnxscript/ir/_external_data_test.py

```diff
@@ -55,5 +59,167 @@ def test_set_base_dir_sets_base_dir_for_all_external_tensors(self):
         self.assertEqual(attr_tensor.base_dir, expected_dir)
 
 
+class OffsetCalcTest(unittest.TestCase):
+    """Test the offset calculation for the external tensor class."""
+
+    def test_align_offset_false(self):
+        # Tensor size > Align Threshold
+        current_offset = 20000
+        tensor_size = 1048
+        new_offset = _external_data.compute_new_offset(
+            current_offset, tensor_size, align_offset=False
+        )
+        self.assertEqual(current_offset, new_offset)
+
+    def test_align_with_small_align_threshold(self):
+        # Tensor size < Align Threshold
+        current_offset = 20000
+        tensor_size = 1048
+        new_offset = _external_data.compute_new_offset(
+            current_offset,
+            tensor_size,
+            align_threshold=1000,
+        )
+        self.assertNotEqual(current_offset, new_offset)
+
+    def test_align_with_large_align_threshold(self):
+        # Tensor size > Align Threshold
+        current_offset = 20000
+        tensor_size = 1048
+        new_offset = _external_data.compute_new_offset(
+            current_offset,
+            tensor_size,
+        )
+        self.assertEqual(current_offset, new_offset)
+
+    def test_allocation_granularity_diff(self):
+        # Tensor size > Align Threshold
+        current_offset = 20000
+        tensor_size = 1048577
+        new_offset_1 = _external_data.compute_new_offset(
+            current_offset,
+            tensor_size,
+            allocation_granularity=4000,
+        )
+        new_offset_2 = _external_data.compute_new_offset(
+            current_offset,
+            tensor_size,
+        )
+        self.assertNotEqual(current_offset, new_offset_1)
+        self.assertNotEqual(current_offset, new_offset_2)
+        self.assertNotEqual(new_offset_1, new_offset_2)
+
+
+class ExternalTensorTest(unittest.TestCase):
+    """Test the memory mapped external tensor class."""
```

Comment:
Is this needed/accurate? Are there new tests we should move to core test?

Reply:
I guess this is needed for now, given the model is built using IR classes as compared to the onnx helpers (in core). Will use this as a base unittest class for adding the tests you mentioned below.

Reply:
There is one here: https://github.com/microsoft/onnxscript/blob/ce7711ea487a91baff2ae97a2db4e83b3ae442bb/onnxscript/ir/_core_test.py#L180.

Should they be combined?